### ISSUE TITLE
Update google_checks.xml to have the SuppressionCommentFilter and SuppressWarningsHolder modules in the config by default (and by extension, SuppressWarningsFilter)

### ISSUE THREAD

**Issue by ddcprg** [checkstyle/checkstyle#11655]
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac MyClass.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat google_checks.xml
...
<module name="Checker">
  <!-- IMPORTANT: Allows use of SuppressWarnings annotation -->
  <module name="SuppressWarningsFilter"/>
  ...
  <module name="TreeWalker">
    <module name="SuppressWarningsHolder" />
    <module name="SuppressionCommentFilter">
      <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)" />
      <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)" />
      <property name="checkFormat" value="$1" />
    </module>
...
</module>


/var/tmp $ cat MyClass.java
class MyClass {
  public void aMethod() {
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.2-all.jar -c google_checks.xml MyClass.java
Starting audit...
[WARN] /var/tmp/MyClass.java:2:15: Method name 'aMethod' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9_]*$'. [MethodName]
Audit done.
```

After using `@SuppressWarnings` 

```
/var/tmp $ cat MyClass.java
class MyClass {
  @SuppressWarnings("checkstyle:methodname")
  public void aMethod() {
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.2-all.jar -c google_checks.xml MyClass.java
Starting audit...
Audit done.
```

Using special comments to switch checkstyle off and on:

```
/var/tmp $ cat MyClass.java
class MyClass {
  // CHECKSTYLE.OFF: MethodName
  public void aMethod() {
  }
  // CHECKSTYLE.ON: MethodName
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.2-all.jar -c google_checks.xml MyClass.java
Starting audit...
Audit done.
```

**Is your feature request related to a problem? Please describe.**
No, this is just additional config in `google_checks.xml`.

**Describe the solution you'd like**
Add the ability to suppress style checks via annotations and special comments around the code. This will offer users alternatives to suppress checks other than using XML.

**Additional context**
Some of these changes have been already adopted by Google https://github.com/GoogleCloudPlatform/java-repo-tools/blob/main/third_party/checkstyle/google_checks.xml

**References**
* http://stackoverflow.com/a/22556386/101923
* http://stackoverflow.com/questions/4023185

**Comment by ddcprg** [comment]
@nick-mancuso I've updated the config to show the relevant bits only

**Comment by nick-mancuso** [comment]
@ddcprg to be clear, you would like google_checks.xml to have the SuppressionCommentFilter and SuppressWarningsHolder modules in the config by default (and by extension, SuppressWarningsFilter), correct?

**Comment by nick-mancuso** [comment]
@ddcprg please update issue title to be more along the lines of https://github.com/checkstyle/checkstyle/issues/11655#issuecomment-1131613563, make sure to mention modules you want and google_checks.xml, please.

**Comment by ddcprg** [comment]
@nick-mancuso thanks, copy&pasted now - almost

**Comment by nick-mancuso** [comment]
Approved, in PR for this issue, we should do:

1. Add SuppressWithNearbyCommentFilter, SuppressionCommentFilter, and SuppressWarningsHolder modules
2. Add SuppressWarningsFilter
3. Update documentation at https://checkstyle.sourceforge.io/google_style.html#Google_Suppressions

**Comment by ddcprg** [comment]
thanks @nick-mancuso @romani I'll include the additional module and fix the PR

**Comment by nick-mancuso** [comment]
Fix was merged.
